homeassistant/components/prometheus/__init__.py

```diff
@@ -394,6 +399,26 @@ def _handle_device_tracker(self, state):
         value = self.state_as_number(state)
         metric.labels(**self._labels(state)).set(value)
 
+        # record battery level of the device tracker if available
+        self._battery(state)
+
+        # device-tracker entities have certain well known attributes that we can record here in prometheus as a
+        # time series.
+        well_known_attributes = {
+            # base state attributes
+            ATTR_LATITUDE: "Latitude of the device tracker measured in decimal degrees",
+            ATTR_LONGITUDE: "Longitude of the device tracker measured in decimal degrees",
+            ATTR_GPS_ACCURACY: "GPS accuracy of the device tracker location",
+        }
+        for attr, description in well_known_attributes.items():
+            value = state.attributes.get(attr)
+            if value is not None and isinstance(value, (int, float)):
+                self._metric(
+                    f"device_tracker_{attr}",
```

Comment:
Seems a bit risky to use the const value as part of the metric name. Right now those consts are reasonably looking lowercase human-friendly strings (like `gps_accuracy`), but I am not sure to what extent we can rely on that remaining in the future? Perhaps set metric names explicitly here, even if will seem a bit repetitive.

Reply:
Thanks @knyar, good idea.